features/323-login-id/stage-2.md

```diff
@@ -0,0 +1,101 @@
+# Multiple login IDs per key, Configurable login ID keys, Standard semantics keys
+
+## Database Schema
+
+- Update `_core_user.verify_info` column:
+    - It is a string map keyed by login ID key,
+      with verified boolean flag as value.
+    - This format is incompatible with multiple login ID per login ID key.
+    - Migrate to keyed for login ID instead.
+    - All user would be treated as NOT verified after migration, and would need
+      to be verified again.
+
+## Configuration
+- Add field `loginIDKeys` (map of string to `LoginIDConfigMap`, optional).
```

Comment:
Does it allows something like this?

```json
{
    "email": {"type": "email"},
    "email-addr": {"type": "email"},
}
```

If yes, it would violate the primary ID spec below. If not, we should make it clear that each type could only be assigned to one key.

Reply:
(Okay, but if each type could only be assigned to one key, why not we just enforce it must be called email, phone and username lol) @louischan-oursky @kiootic

Reply:
Multiple key per type should be supported, I'll clarify how it interact with the welcome email feature.